packages/@n8n/nodes-langchain/nodes/agents/Agent/agents/utils.ts

```diff
@@ -17,3 +18,24 @@ export async function extractParsedOutput(
 	// with fallback to the original output if it's not present
 	return parsedOutput?.output ?? parsedOutput;
 }
+
+export async function checkForStructuredTools(
+	tools: Tool[],
+	node: INode,
+	currentAgentType: string,
+) {
+	const dynamicStructuredTools = tools.filter(
+		(tool) => tool.constructor.name === 'DynamicStructuredTool',
```

Comment:
Would it be more reliable to check `instanceof` here?

Reply:
It would be less reliable :D Because then it would be compared against a specific imported instance of that class. So if the import comes from a different langchain package or version, it wouldn't trigger this condition

Reply:
Fair enough :D But I worry about the case if we introduce some class extending `DynamicStructuredTool` in the future for example. Should pop up in tests, I guess, so feel free to disregard my comment.